swap/src/monero/wallet_rpc.rs

```diff
@@ -211,9 +215,13 @@ impl WalletRpc {
             file.flush().await?;
         }
 
-        let file = std::fs::File::open(monero_wallet_rpc.archive_path())?;
+        let archive_path = monero_wallet_rpc.archive_path();
+        let unzip: JoinHandle<Result<ZipArchive<File>>> = tokio::spawn(async {
+            let file = File::open(archive_path)?;
+            Ok(ZipArchive::new(file)?)
```

Comment:
1. That is not gonna do much, you will need to move all IO operations into the task.
2. You need to use `spawn_blocking` otherwise there is practically no difference to just doing it inline.